lib/fs.js

```diff
@@ -395,12 +395,25 @@ function readFileAfterClose(err) {
   else
     buffer = context.buffer;
 
-  if (context.encoding)
-    buffer = buffer.toString(context.encoding);
+  if (err) return callback(err, buffer);
 
-  callback(err, buffer);
+  if (context.encoding) {
+    return tryToString(buffer, context.encoding, callback);
+  }
+
+  callback(null, buffer);
 }
 
+function tryToString(buf, encoding, callback) {
+  var e;
+  try {
+    buf = buf.toString(encoding);
+  }
+  catch (err) {
```

Comment:
style? i thought the linter would catch this. catch should be on the same line as `}`.